docs/frameworks/index.rst

```diff
@@ -0,0 +1,55 @@
+.. _spec:
+
+=======================
+Framework specification
+=======================
+
+The role of web-poet is to define a standard on how to write the extraction
+logic for a webpage, and allow it to be reused in different web scraping
+frameworks.
+
+:ref:`Page objects <page-objects>` should be flexible enough to be used with:
+
+* synchronous or asynchronous frameworks, callback-based and
+  ``async def / await`` based,
+* single-node and distributed systems,
+* different underlying HTTP implementations - or without HTTP support
+  at all, etc.
+
+To support web-poet in a web scraping framework, you need to:
+
+-   Support returning a :ref:`page object <page-objects>` based on a given
+    target URL and a desired :ref:`output item type <items>`.
+
+    To do that:
+
+    -   Use ``web_poet.default_registry`` to determine the right :ref:`page
+        object class <page-object-classes>` to use.
+
+        Optionally, let users indicate their own
+        :class:`~web_poet.rules.RulesRegistry` object.
+
+    -   Support building :ref:`input objects <inputs>` based on type hints,
+        i.e. dependency injection.
+
+        You must support all input classes from the :mod:`web_poet.page_inputs`
+        module.
```

Comment:
I don't think it's a requirement. scrapy-poet doesn't support all input classes from page_inputs. If something is not supported, a framework should raise an error.